### PR TITLE
[vSphere] Use envtest for testing machine patch

### PR DESCRIPTION
**Pull request by alexander-demichev**
This PR introduces the usage of envtest for testing machine patch

### CODE REVIEW

File: pkg/controller/vsphere/machine_scope_test.go [hunk 1]
```diff
@@ -276,11 +282,34 @@ func TestGetCredentialsSecret(t *testing.T) {
 }
 
 func TestPatchMachine(t *testing.T) {
+	g := NewWithT(t)
+
+	configv1.AddToScheme(scheme.Scheme)
+
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{filepath.Join("..", "..", "..", "config", "crds")},
```
Comment (enxebre) [inline pkg/controller/vsphere/machine_scope_test.go:290]: we should probably rely on the CRDs on `install/` and drop the `config` folder. It was originally used for kubemark testing which we don't run anymore.

File: pkg/controller/vsphere/machine_scope_test.go [hunk 2]
```diff
@@ -276,11 +282,34 @@ func TestGetCredentialsSecret(t *testing.T) {
...
+	}
+
+	cfg, err := testEnv.Start()
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(cfg).ToNot(BeNil())
+	defer func() {
+		g.Expect(testEnv.Stop()).To(Succeed())
+	}()
+
+	k8sClient, err := client.New(cfg, client.Options{})
```
Comment (enxebre) [inline pkg/controller/vsphere/machine_scope_test.go:300]: You'll need to pass `Scheme: scheme.Scheme` here to make the client aware of the APIs

**Comment by enxebre** [comment]
thanks! dropped a few comments, otherwise looks great!

File: pkg/controller/vsphere/machine_scope_test.go (hunk 2 above)
Comment (JoelSpeed) [reply pkg/controller/vsphere/machine_scope_test.go:304]: Does this need addressing?

Comment (alexander-demichev) [reply pkg/controller/vsphere/machine_scope_test.go:304]: No, client.New() will use default scheme

**Pull request merged**